From: =?utf-8?q?Jonas_=C3=85dahl?= <jadahl@gmail.com>
Date: Thu, 14 Jan 2021 16:06:45 +0100
Subject: clutter/seat: Don't unref backend on finalize

We never took a ref on the backend, it's the backend that owns the seat,
not the other way around.

This silences warnings spewed on tear down.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1670>
---
 clutter/clutter/clutter-seat.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/clutter/clutter/clutter-seat.c b/clutter/clutter/clutter-seat.c
index 4af7e28..5799c20 100644
--- a/clutter/clutter/clutter-seat.c
+++ b/clutter/clutter/clutter-seat.c
@@ -113,17 +113,6 @@ clutter_seat_get_property (GObject    *object,
     }
 }
 
-static void
-clutter_seat_finalize (GObject *object)
-{
-  ClutterSeat *seat = CLUTTER_SEAT (object);
-  ClutterSeatPrivate *priv = clutter_seat_get_instance_private (seat);
-
-  g_clear_object (&priv->backend);
-
-  G_OBJECT_CLASS (clutter_seat_parent_class)->finalize (object);
-}
-
 static void
 clutter_seat_class_init (ClutterSeatClass *klass)
 {
@@ -131,7 +120,6 @@ clutter_seat_class_init (ClutterSeatClass *klass)
 
   object_class->set_property = clutter_seat_set_property;
   object_class->get_property = clutter_seat_get_property;
-  object_class->finalize = clutter_seat_finalize;
 
   signals[DEVICE_ADDED] =
     g_signal_new (I_("device-added"),
